daemon/container_operations.go

```diff
@@ -691,6 +691,12 @@ func (daemon *Daemon) connectToNetwork(cfg *config.Config, container *container.
 	}
 	nwName := n.Name()
 
+	if idOrName != container.HostConfig.NetworkMode.NetworkName() {
```

Comment:
I really don't like the idea of mutating the network mode in `connectToNetwork` but there's no other choice: if the network mode references a swarm-scope network, it might not exist on the local node before `findAndAttachNetwork` is executed. As a Swarm agent doesn't have access to the list of swarm-scope networks, this is the only place where `normalizeNetMode` can be called.

As discussed with @thaJeztah today, I think it'd be preferable to create uninitialized/placeholder swarm-scope networks on every node. That would make the UX better and it'd restore the original semantics (ie. a container can't connect to a network that don't exist locally). That would make the code easier to reason about, we'd be able to move the call to `normalizeNetMode` upper in the call stack, and we'd not need to create a Swarm task in `findAndAttachNetwork` (ie. this too could be moved upper in the stack).